og.api.php

```diff
@@ -0,0 +1,102 @@
+  // builders to toggle a configuration setting that will prevent groups to be
+  // deleted if they are published.
+  // Retrieve the module configuration.
+  $config = \Drupal::config('mymodule.settings');
+
+  // Check if the site is configured to allow deletion of published groups.
+  $published_groups_can_be_deleted = \Drupal::config('mymodule')->get('delete_published_groups');
+
+  // If deletion is not allowed and the group is published, revoke the
+  // permission.
+  $group = $context['group'];
+  if ($group instanceof EntityPublishedInterface && !$group->isPublished() && !$published_groups_can_be_deleted) {
+    $key = array_search(OgAccess::DELETE_GROUP_PERMISSION, $permissions);
+    if ($key !== FALSE) {
+      unset($permissions[$key]);
+    }
+  }
+
+  // Since our access result depends on our custom module configuration, we need
+  // to add it to the cache metadata.
+  $cacheable_metadata->addCacheableDependency($config);
+}
+
+/**
+ * Allows to alter access to entity operations performed on group content.
+ *
+ * @param \Drupal\Core\Access\AccessResultInterface $access_result
+ *   The access result being altered.
+ * @param \Drupal\Core\Cache\CacheableMetadata $cacheable_metadata
+ *   The cache metadata, passed by reference.
+ * @param array $context
+ *   An associative array containing contextual information, with keys:
+ *   - 'operation': The entity operation being performed on the group content.
+ *   - 'group': The group entity to which the group content belongs.
+ *   - 'group_content': The group content entity upon which the operation is
+ *      performed.
+ *   - 'user': The user account for which access is being determined.
+ */
+function hook_og_user_access_entity_operation_alter(AccessResultInterface &$access_result, CacheableMetadata &$cacheable_metadata, array $context): void {
+  // This example implements a use case where a custom module allows site
+  // builders to toggle a configuration setting that will allow users with the
+  // site wide 'edit and delete comments in all groups' permission to edit and
+  // delete all comments in all groups, even if they are not a group member.
+  /** @var \Drupal\Core\Session\AccountProxyInterface $user */
+  $user = $context['user'];
+  $group_content = $context['group_content'];
+
+  // Retrieve the module configuration.
+  $config = \Drupal::config('mymodule.settings');
+
+  // If comment moderation is allowed and the user has the permission, grant
+  // access to the 'update' and 'delete' operations on comment entities.
+  $is_comment = $group_content->getEntityTypeId() === 'comment';
+  $user_can_moderate_comments = $user->hasPermission('edit and delete comments in all groups');
+  $comment_moderation_is_enabled = \Drupal::config('mymodule')->get('comment_moderation_enabled');
+
+  if ($is_comment && $user_can_moderate_comments && $comment_moderation_is_enabled) {
+    $access_result = new AccessResultAllowed();
```

Comment:
Probably just a matter of taste:
```suggestion
    $access_result = AccessResult::allowed();
```